docs/proposal/AI-gateway.md

```diff
@@ -0,0 +1,206 @@
+
+2. Specific AI traffic management policy: such as rate limit, cache, fallback, etc. Allow users to configure any number and type of policies.
+
+API:
+
+```go
+// AIExtension is the Schema for the aiextensions API.
+type AIExtension struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AIExtensionSpec   `json:"spec,omitempty"`
+	Status AIExtensionStatus `json:"status,omitempty"`
+}
+
+// AIExtensionSpec defines the desired state of AIExtension.
+type AIExtensionSpec struct {
+	Options Options `json:"options"`
+}
+
+type Options struct {
+	RateLimits []RateLimit `json:"rateLimits,omitempty"`
+  // Add more polices as needed.
+}
+
+// +kubebuilder:validation:Enum=second;minute;hour;day;month
+type RateLimitUnit string
+
+const (
+	Second RateLimitUnit = "second"
+	Minute RateLimitUnit = "minute"
+	Hour   RateLimitUnit = "hour"
+	Day    RateLimitUnit = "day"
+	Month  RateLimitUnit = "month"
+)
+
+// RateLimit Policy for AI traffic.
+type RateLimit struct {
+	RequestsPerUnit uint32        `json:"requestsPerUnit"`
+	Unit            RateLimitUnit `json:"unit"`
+	Model           string        `json:"model"`
+}
+
+// AIExtensionStatus defines the observed state of AIExtension.
+type AIExtensionStatus struct {
+}
+```
+
+Only the `RateLimit` policy config is shown above. The data structure of new polices can be added directly to the `Options` structure as needed.
+
+The config example of the above Go code API converted into yaml format is as follows:
+
+```yaml
+apiVersion: ai.kmesh.net/v1alpha1
+kind: AIExtension
+metadata:
+  name: ai-extensions
+  namespace: ollama
+spec:
+  targetRefs:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      name: ollama
+  options:
+    rateLimits:
+      - requestsPerUnit: 10
+        unit: minute
+        model: llama3.2
+```
+
+In accordance with the Gateway API convention, the service is associated with `HTTPRoute` in `targetRefs`, and then the specific AI policies are configured under `options`. Currently all AI policies are configured in the `AIExtension` CRD. If necessary, separate CRDs can be built for different policies in the future.
+
+#### Integration with Existing Gateways
+
+The integration of various existing gateways and AI plugins is also very convenient. For example, Envoy Gateway provides a dedicated CRD for External Processing. As follows:
+
+```yaml
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: EnvoyExtensionPolicy

```

Comment:
how can integrate with istio ingress gateway?

Reply:
EnvoyFilter is required. If necessary, example could be listed here as well.